### PR TITLE
gr-vocoder: Removed 'snprintf' usage

### PR DESCRIPTION
**Pull request by alekhgupta1441**
Fixes: #2875 Removed 'snprintf' usage from gr-vocoder

### CODE REVIEW

**Comment by drmpeg** [comment]
Your commit is failing the CheckPR Formatting step because you have a space at the end of the line:
```
    strcpy(d_cb_state.tx_str, txstr.c_str());
```
The utility that's being run by Check PR Formatting is clang-format.

Also, you misspelled gr-vocoder in the commit message.

**Comment by alekhgupta1441** [comment]
@drmpeg Thanks for telling my mistake. I was still wondering on this.
I will soon make changes but I am unable to do this for a while as my system has got some hardware damage recently. I will do this as early as possible.

File: gr-vocoder/lib/freedv_tx_ss_impl.cc [hunk 1]
```diff
@@ -70,7 +70,8 @@ freedv_tx_ss_impl::freedv_tx_ss_impl(int mode,
     if ((d_freedv = freedv_open(mode)) == NULL)
         throw std::runtime_error("freedv_tx_ss_impl: freedv_open failed");
 #endif
-    snprintf(d_cb_state.tx_str, 79, "%s", d_msg_text.c_str());
+    std::string txstr = str(boost::format("%s") % d_msg_text.c_str());
```
Comment (mbr0wn) [inline gr-vocoder/lib/freedv_tx_ss_impl.cc:73]: One of these is not like the other: The left hand has the 79-char limit, the right hand does not, and `txstr` serves no purpose, nor does `boost::format()`.

Comment (ThomasHabets) [reply gr-vocoder/lib/freedv_tx_ss_impl.cc:73]: Why not replace with `auto txstr = d_msg_text.substr(0,79)`?

But also, what's so bad about `snprintf()`? Seems like a safe way to cap a string and always have it be null-terminated.

Comment (ThomasHabets) [reply gr-vocoder/lib/freedv_tx_ss_impl.cc:73]: In terms of "code smell", `strcpy` is much more pungent than `snprintf`.

**Comment by mbr0wn** [comment]
Agree with @ThomasHabets. This does not seem like an improvement. Also, we don't like to use Boost if we don't have to.

**Pull request closed without merging**